kernel/src/engine/arrow_expression.rs

```diff
@@ -867,4 +948,205 @@ mod tests {
+            .downcast::<ArrowEngineData>()
+            .unwrap()
+            .into();
+        assert_eq!(actual_rb, expected);
+    }
+
+    #[test]
+    fn test_create_one_disallow_column_ref() {
+        let expr = Expression::struct_from([column_expr!("a")]);
+        let schema = Arc::new(crate::schema::StructType::new([StructField::new(
+            "a",
+            DeltaDataTypes::INTEGER,
+            true,
+        )]));
+
+        let handler = ArrowExpressionHandler;
+        let actual = handler.create_one(schema, &expr);
+        assert!(actual.is_err());
+    }
+
+    #[test]
+    fn test_create_one_disallow_operator() {
+        let expr = Expression::struct_from([Expression::binary(
+            BinaryOperator::Plus,
+            Expression::literal(1),
+            Expression::literal(2),
+        )]);
+        let schema = Arc::new(crate::schema::StructType::new([StructField::new(
+            "a",
+            DeltaDataTypes::INTEGER,
+            true,
+        )]));
+
+        let handler = ArrowExpressionHandler;
+        let actual = handler.create_one(schema, &expr);
+        assert!(actual.is_err());
+    }
+
+    #[test]
+    fn test_create_one_nested() {
+        let expr = Expression::struct_from([Expression::struct_from([
+            Expression::literal(1),
+            Expression::literal(2),
+        ])]);
+        let schema = Arc::new(crate::schema::StructType::new([StructField::new(
+            "a",
+            DeltaDataTypes::struct_type([
+                StructField::new("b", DeltaDataTypes::INTEGER, true),
+                StructField::new("c", DeltaDataTypes::INTEGER, false),
+            ]),
+            false,
+        )]));
+
+        let handler = ArrowExpressionHandler;
+        let actual = handler.create_one(schema, &expr).unwrap();
+        let expected_schema = Arc::new(arrow_schema::Schema::new(vec![Field::new(
+            "a",
+            DataType::Struct(
+                vec![
+                    Field::new("b", DataType::Int32, true),
+                    Field::new("c", DataType::Int32, false),
+                ]
+                .into(),
+            ),
+            false,
+        )]));
+        let expected = RecordBatch::try_new(
+            expected_schema,
+            vec![Arc::new(StructArray::from(vec![
+                (
+                    Arc::new(Field::new("b", DataType::Int32, true)),
+                    create_array!(Int32, [1]) as ArrayRef,
+                ),
+                (
+                    Arc::new(Field::new("c", DataType::Int32, false)),
+                    create_array!(Int32, [2]) as ArrayRef,
+                ),
+            ]))],
+        )
+        .unwrap();
+        let actual_rb: RecordBatch = actual
+            .into_any()
+            .downcast::<ArrowEngineData>()
+            .unwrap()
+            .into();
+        assert_eq!(actual_rb, expected);
+
+        // make the same but with literal struct instead of struct of literal
```

Comment:
Hmm I think we were discussing expression list  struct and struct literal before. iirc we weren't happy with these two concepts. It's once again weird and confusing 🤔 

The schema is the same, but the expression is different. I wouldn't block this PR on that, but I wanted to flag this.